extensions/amp-a4a/0.1/test/test-a4a-integration.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    MockA4AImpl,
+    stringToArrayBuffer,
+    SIGNATURE_HEADER,
+    TEST_URL,
+} from './utils';
+import {Xhr} from '../../../../src/service/xhr-impl';
+import {createIframePromise} from '../../../../testing/iframe';
+import {
+    data as validCSSAmp,
+} from './testdata/valid_css_at_rules_amp.reserialized';
+import {installDocService} from '../../../../src/service/ampdoc-impl';
+import {adConfig} from '../../../../ads/_config';
+import {a4aRegistry} from '../../../../ads/_a4a-config';
+import {
+    resetScheduledElementForTesting,
+    upgradeOrRegisterElement,
+} from '../../../../src/custom-element';
+import * as sinon from 'sinon';
+
+chai.Assertion.addMethod('renderedInFriendlyIframe', function(srcdoc) {
+  const obj = this._obj;
+  this.assert(obj,
+      'amp ad element should be #{exp}, got #{act}',  // if true message
+      'amp ad element should not be #{exp}, got #{act}',  // if negated message
+      'truthy',  // expected
+      obj);      // actual
+  const elementName = obj.tagName.toLowerCase();
+  const child = obj.querySelector('iframe[srcdoc]');
+  this.assert(child,
+      `child of amp ad element ${elementName} should be #{exp}, got #{act}`,
+      `child of amp ad element ${elementName} should not be #{exp}, got #{act}`,
+      'truthy',
+      child);
+  this.assert(child.getAttribute('srcdoc').indexOf(srcdoc) >= 0,
+      `iframe child of amp ad element ${elementName} should contain #{exp}, ` +
+      'was #{act}',
+      `iframe child of amp ad element ${elementName} should not contain ` +
+      '#{exp}, was #{act}',
+      srcdoc,
+      child.getAttribute('srcdoc'));
+  const childBody = child.contentDocument.body;
+  this.assert(childBody,
+      `iframe child of amp ad element ${elementName} should have #{exp}`,
+      `iframe child of amp ad element ${elementName} should have #{exp}, ` +
+      'got #{act}',
+      'body tag',
+      childBody);
+  [obj, child, childBody].forEach(toTest => {
+    expect(toTest).to.be.visible;
+  });
+});
+
+chai.Assertion.addMethod('renderedInXDomainIframe', function(src) {
```

Comment:
maybe just make it a local function? this add a method to global chai, it's the same effect as you add it in  `_init_tests.js`